server/route.go

```diff
@@ -423,9 +423,12 @@ func (s *Server) routeSidQueueSubscriber(rsid []byte) (*subscription, bool) {
 	}
 	sid := matches[RSID_SID_INDEX]
 
+	client.mu.Lock()
 	if sub, ok := client.subs[string(sid)]; ok {
+		client.mu.Unlock()
```

Comment:
Let's clean this up and just have one unlock.
So capture sub and ok then unlock, then return as appropriate.